src/explore-education-statistics-common/src/modules/find-statistics/components/charts/AbstractChart.tsx

```diff
@@ -0,0 +1,126 @@
+import { CharacteristicsData } from '@common/services/tableBuilderService';
+import { Axis, ReferenceLine } from '@common/services/publicationService';
+import { Component, ReactNode } from 'react';
+import React from 'react';
+import * as Recharts from 'recharts';
+import { PositionType } from 'recharts';
+import { Label } from 'recharts';
+import { XAxisProps } from 'recharts';
+import { XAxis } from 'recharts';
+import { YAxisProps } from 'recharts';
+import { YAxis } from 'recharts';
+
+export interface ChartProps {
+  characteristicsData: CharacteristicsData;
+  chartDataKeys: string[];
+  labels: { [key: string]: string };
+  xAxis: Axis;
+  yAxis: Axis;
+  height?: number;
+  referenceLines?: ReferenceLine[];
+}
+
+export class AbstractChart<P extends ChartProps, S = {}> extends Component<
```

Comment:
Inheritance is normally discouraged for React (https://reactjs.org/docs/composition-vs-inheritance.html). Wonder if we could just have these methods as chart util functions in some standalone module for easier re-use.

Reply:
What reuse? At present the only reuse is the charts. There is severe dependency on knowledge of the chart properties.

Reply:
Rather than functions, I should reword it as, it actually feels like we've got custom components in here that we can split out.

- `calculateXAxis` and `calculateYAxis` could be `XAxis` and `YAxis` components.
- `calculateMargins` could be in some margin util module.
- `generateReferenceLines` is a `ReferenceLines` component.

Reply:
Hey I was thinking that too... but this was a prototype and it was quicker to get this done.